src/LensFactory.m

```diff
@@ -9,7 +9,9 @@
     end
 
     function t = build_triangle(x, y, dims)
-      t = [];
+      s = Triangle(Vec(x, y));
+      s.set_dimensions(dims);
+      t = Lens(s);
     end
```

Comment:
Also write the factory methods in MirrorFactory, BlackBodyFactory and TargetFactory